src/main/java/org/apache/maven/plugin/resources/remote/ProcessRemoteResourcesMojo.java

```diff
@@ -1336,6 +1341,41 @@ else if ( appendedVmResourceFile.exists() )
         }
     }
 
+    private void copyBundleResource( ClassLoader classLoader, String bundleResource, File to )
+        throws IOException
+    {
+        URL resUrl = classLoader.getResource( bundleResource );
+        if ( resUrl != null )
+        {
+            FileUtils.copyURLToFile( resUrl, to );
+        }
+    }
+
+    private void filterBundleResource( ClassLoader classLoader, String bundleResource, File to )
+        throws IOException, MojoExecutionException
+    {
+        File tmpFile = null;
+        try
+        {
+            tmpFile = File.createTempFile( "maven-remote-resources-plugin", null );
+            tmpFile.deleteOnExit();
```

Comment:
Consider moving `deleteOnExit()` to the finally, where the file will either be deleted at once, or a warning will be written.
`deleteOnExit()` adds the `Path` to a list of files to delete on exit and does not scale well if there are many files.

Reply:
Nice, I didn't realise this! It's fixed in 251f5d3bf6b0290a25d4c7f21e9d2430fdc5837e.